pkg/helm: set wait to true

invidian opened the pull request:
So when installing the chart, bootkube will wait for the chart to
converge before returning. This is to ensure, that everything has been
deployed correctly.

Closes #8

This is currently problematic, as in Lokomotive case, the Calico charts gets installed before Kubernetes chart, so Calico chart never converges, as the nodes are not able to join the cluster as Kubernetes chart is missing. The potential solution for that would be to e.g. install the charts in parallel.

Signed-off-by: Mateusz Gozdek <mateusz@kinvolk.io>

ipochi commented:
> So when installing the chart, bootkube will wait for the chart to
> converge before returning. This is to ensure, that everything has been
> deployed correctly.
> 
> Closes #8
> 
> This is currently problematic, as in Lokomotive case, the Calico charts gets installed before Kubernetes chart, so Calico chart never converges, as the nodes are not able to join the cluster as Kubernetes chart is missing. The potential solution for that would be to e.g. install the charts in parallel.
> 
> Signed-off-by: Mateusz Gozdek [mateusz@kinvolk.io](mailto:mateusz@kinvolk.io)

Have we encountered such scenarios in our CI pipeline where a Calico chart never converges.


Also If you could please tag the lokomotive PR link so that we know that the CI job has been run on these changes.

invidian commented:
>Have we encountered such scenarios in our CI pipeline where a Calico chart never converges.

Yes, this is why I had to make it parallel. I don't have a link to the CI job unfortunately (though it is not very useful anyway probably).

>Also If you could please tag the lokomotive PR link so that we know that the CI job has been run on these changes.

The changes has been tested here: https://github.com/kinvolk/lokomotive/pull/805. And they indeed works fine.

invidian commented:
I converted this PR from draft, to trigger merge conflicts check, as even though I rebased the PR it was still showing as conflicting and it seems, that there is now way to convert it back to draft now :/ Anyway, this can be merged, but cannot be updated in Lokomotive.

invidian commented:
As we migrated to Kubernetes 1.19 in Lokomotive, CI is now passing with this change :tada:: https://github.com/kinvolk/lokomotive/pull/1085
